rcp/org.testeditor.rcp4.views.projectexplorer/src/org/testeditor/rcp4/views/projectexplorer/ProjectExplorer.xtend

```diff
@@ -11,15 +11,53 @@
  */
 package org.testeditor.rcp4.views.projectexplorer
 
+import org.eclipse.core.resources.IFolder
+import org.eclipse.core.resources.IResourceChangeEvent
+import org.eclipse.core.resources.ResourcesPlugin
+import org.eclipse.jdt.core.IClasspathEntry
+import org.eclipse.jdt.core.JavaCore
+import org.eclipse.jdt.internal.core.JavaProject
+import org.eclipse.jface.viewers.LabelProviderChangedEvent
 import org.eclipse.swt.widgets.Composite
+import org.eclipse.ui.PlatformUI
 import org.eclipse.ui.navigator.CommonNavigator
 
 class ProjectExplorer extends CommonNavigator {
-	
+
 	override void createPartControl(Composite parent) {
 		super.createPartControl(parent)
 		// TODO Make this configurable with a new Configuration dialog.
 		setLinkingEnabled(true)
+
+		ResourcesPlugin.workspace.addResourceChangeListener([resourcePostChangeSubscription],
+			IResourceChangeEvent.POST_CHANGE);
+	}
+
+	def void resourcePostChangeSubscription(IResourceChangeEvent data) {
```

Comment:
This looks like a simple listener. If so it should be extracted to a separate class.